python_ta/transforms/type_inference_visitor.py

```diff
@@ -234,10 +234,18 @@ def visit_compare(self, node):
     def visit_assign(self, node):
         # multi-assignment; LHS is a tuple of AssignName target nodes as "elements"
         if isinstance(node.targets[0], astroid.Tuple):
-            target_type_tuple = zip(node.targets[0].elts, node.value.elts)
-            for target_node, value in target_type_tuple:
-                target_type_var = node.frame().type_environment.lookup_in_env(target_node.name)
-                self.type_constraints.unify(target_type_var, value.type_constraints.type)
+            if isinstance(node.value, astroid.Name):
```

Comment:
This is not a good choice of special case.

The real special case should be `isinstance(node.value, astroid.Tuple)`. For everything else, lookup the type and then unify_call with `__iter__`.